drivers/dai/intel/dmic/dmic_nhlt.c

```diff
@@ -125,6 +126,69 @@ static int dai_nhlt_dmic_dai_params_get(struct dai_intel_dmic *dmic,
 
 	return 0;
 }
+
+
+/*
+ * @brief Set clock source used by device
+ *
+ * @param source Clock source index
+ */
+static inline void dai_dmic_clock_select_set(const struct dai_intel_dmic *dmic, uint32_t source)
+{
+	uint32_t val;
+#ifdef CONFIG_SOC_INTEL_ACE20_LNL /* Ace 2.0 */
+	val = sys_read32(dmic->vshim_base + DMICLVSCTL_OFFSET);
+	val &= ~DMICLVSCTL_MLCS;
+	val |= FIELD_PREP(DMICLVSCTL_MLCS, source);
+	sys_write32(val, dmic->vshim_base + DMICLVSCTL_OFFSET);
+#else
+	val = sys_read32(dmic->shim_base + DMICLCTL_OFFSET);
+	val &= ~DMICLCTL_MLCS;
+	val |= FIELD_PREP(DMICLCTL_MLCS, source);
+	sys_write32(val, dmic->shim_base + DMICLCTL_OFFSET);
+#endif
+}
+
+/*
+ * @brief Get clock source used by device
+ *
+ * @return Clock source index
+ */
+static inline uint32_t dai_dmic_clock_select_get(const struct dai_intel_dmic *dmic)
+{
+	uint32_t val;
+#ifdef CONFIG_SOC_INTEL_ACE20_LNL /* Ace 2.0 */
+	val = sys_read32(dmic->vshim_base + DMICLVSCTL_OFFSET);
+	val &= ~DMICLCTL_MLCS;
```

Comment:
this doesn't look right - you're clearing the bits that you're extracting below. You could just `return 0` instead

Reply:
good catch!